Backend/controllers/user.js

```diff
@@ -1,219 +1,291 @@
-const user = require('../models/user')
-const bcrypt = require('bcryptjs')
-const {generateAccessToken, generateRefreshToken, getUser} = require('../service/auth')
-const JWT_SECRET = process.env.JWT_SECRET;  
-
+const user = require("../models/user");
+const bcrypt = require("bcryptjs");
+const {
+  generateAccessToken,
+  generateRefreshToken,
+  getUser,
+} = require("../service/auth");
+const JWT_SECRET = process.env.JWT_SECRET;
+const jwt = require("jsonwebtoken");
+const nodemailer = require("nodemailer");
 
 async function handleUserSignUp(req, res) {
+  try {
     const { name, email, mobile, password } = req.body;
 
-    try {
-        const existingUser = await user.findOne({ email }) || await user.findOne({ mobile });
-        if (existingUser) {
-            return res.status(400).json({ msg: "User already exists" });
-        }
-
-        const passwordValidationRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
-        const nameValidationRegex=/^[a-zA-Zs]+$/;
-        const mobileValidationRegex=/[6-9][0-9]{9}$/;
-
-        if (!passwordValidationRegex.test(password)) {
-            return res.status(400).json({ 
-                msg: "Invalid Password"
-            });
-        }
-
-        if (!nameValidationRegex.test(name)) {
-            return res.status(400).json({ 
-                msg: "Invalid Name"
-            });
-        }
-
-        if (!mobileValidationRegex.test(mobile)) {
-            return res.status(400).json({ 
-                msg: "Invalid Mobile number"
-            });
-        }
-
-        const newUser = await user.create({
-            name,
-            email,
-            mobile,
-            password,
-        });
-
-        return res.json({ msg: "User created", user: newUser });
-    } catch (error) {
-        console.error("Error signing up user:", error);
-        return res.status(500).json({ msg: "Internal server error" });
+    if (!email || !name || !mobile || !password) {
+      return res.status(403).json({ msg: "Please fill your credentials" });
     }
-}
-
 
-async function handleUserLogin(req, res, next) {
-    try {
-        const {email, password} = req.body;
-        const curUser = await user.findOne({email});
-        if(!curUser) {
-            return res.status(400).json({msg: "Invalid Email"});
-        }
+    const existingUser =
```

Comment:
You can write this expression as const existingUser = await user.findOne({
  $or: [{ email }, { mobile }]
});